pkg/metricsadapter/provider/resourcemetrics.go

```diff
@@ -663,7 +631,23 @@ func metricsConvertV1beta1NodeToInternalNode(objs ...unstructured.Unstructured)
 	return nodeMetricsInternal, nil
 }
 
-// generateNamespaceNameKey generates namespace/name key
-func generateNamespaceNameKey(namespace, name string) string {
-	return namespace + "/" + name
+func setMetricsListAnnotationWithClusterName(metricsList *unstructured.UnstructuredList, clusterName string) {
+	if metricsList == nil {
+		return
+	}
+	for i := range metricsList.Items {
+		setMetricsValueAnnotationWithClusterName(&metricsList.Items[i], clusterName)
+	}
+}
+
+func setMetricsValueAnnotationWithClusterName(metricsValue *unstructured.Unstructured, clusterName string) {
+	if metricsValue == nil {
+		return
+	}
+	annotations := metricsValue.GetAnnotations()
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+	annotations[autoscalingv1alpha1.QuerySourceAnnotationKey] = clusterName
+	metricsValue.SetAnnotations(annotations)
 }
```

Comment:
We have a common function here:
https://github.com/karmada-io/karmada/blob/b4cf61419b8228efe73518e608a6fa2ca8ded501/pkg/util/annotation.go#L13-L24

Reply:
Updated.